### PR TITLE
Internationalize log messages

### PR DESCRIPTION
**Pull request by rpjohnst**

### CODE REVIEW

File: BoostTestAdapter/Boost/Runner/BoostTestRunnerBase.cs [hunk 1]
```diff
@@ -151,11 +151,11 @@ private static void KillProcessIncludingChildren(Process process)
             // Killing the main process
             if (KillProcess(process))
             {
-                Logger.Error("Successfully killed process {0}.", process.Id);
+                Logger.Error(Resources.KilledProcess, process.Id);
```
Comment (LukaszMendakiewicz) [inline BoostTestAdapter/Boost/Runner/BoostTestRunnerBase.cs]: Could we here and below use "terminate process" verbiage instead?

File: BoostTestAdapter/BoostTestExecutor.cs [hunk 2]
```diff
@@ -346,20 +350,20 @@ private void RunBoostTests(IEnumerable<TestRun> testBatches, IRunContext runCont
...
                     using (TemporaryFile log    = new TemporaryFile(batch.Arguments.LogFile))
                     using (TemporaryFile stdout = new TemporaryFile(batch.Arguments.StandardOutFile))
                     using (TemporaryFile stderr = new TemporaryFile(batch.Arguments.StandardErrorFile))
                     {
-                        Logger.Debug("Working directory: {0}", batch.Arguments.WorkingDirectory ?? "(null)");
-                        Logger.Debug("Report file      : {0}", batch.Arguments.ReportFile);
-                        Logger.Debug("Log file         : {0}", batch.Arguments.LogFile);
-                        Logger.Debug("StdOut file      : {0}", batch.Arguments.StandardOutFile ?? "(null)");
-                        Logger.Debug("StdErr file      : {0}", batch.Arguments.StandardErrorFile ?? "(null)");
+                        Logger.Debug(Resources.WorkingDirectory, batch.Arguments.WorkingDirectory ?? "(null)");
```
Comment (LukaszMendakiewicz) [inline BoostTestAdapter/BoostTestExecutor.cs:360]: Resx comment on these strings should indicate that we want to have them aligned to each other.

File: BoostTestAdapter/Resources.resx [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+  </data>
+  <data name="WorkingDirectory" xml:space="preserve">
+    <value>Working directory: {0}</value>
+    <comment>{0} represents a file path</comment>
+  </data>
+  <data name="WorkingDirectoryNotFound" xml:space="preserve">
+    <value>Could not retrieve WorkingDirectory from Visual Studio Configuration</value>
+  </data>
+  <data name="WorkingDirectoryNotFoundEx" xml:space="preserve">
+    <value>Could not retrieve WorkingDirectory from Visual Studio Configuration-{0}</value>
```
Comment (LukaszMendakiewicz) [inline BoostTestAdapter/Resources.resx]: Shouldn't "-" rather be ": "?

File: BoostTestAdapter/Resources.resx [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+  <data name="UnableToSetInformation" xml:space="preserve">
+    <value>Unable to set information. Error: {0}</value>
+    <comment>{0} represents a Win32 error</comment>
+  </data>
+  <data name="WorkingDirectory" xml:space="preserve">
+    <value>Working directory: {0}</value>
+    <comment>{0} represents a file path</comment>
+  </data>
+  <data name="WorkingDirectoryNotFound" xml:space="preserve">
+    <value>Could not retrieve WorkingDirectory from Visual Studio Configuration</value>
```
Comment (LukaszMendakiewicz) [inline BoostTestAdapter/Resources.resx]: Lower-case configuration (and below)?

File: BoostTestAdapter/Resources.resx [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+    <comment>{0} represents an XML tag name
+{1} represents an exception message</comment>
+  </data>
+  <data name="KilledProcess" xml:space="preserve">
+    <value>Successfully killed process {0}.</value>
+    <comment>{0} represents a process ID</comment>
+  </data>
+  <data name="ListContentsFile" xml:space="preserve">
+    <value>list_content file: {0}</value>
+    <comment>{0} represents a file path</comment>
```
Comment (LukaszMendakiewicz) [inline BoostTestAdapter/Resources.resx]: Please indicate in the comment that "list_content" is a Boost.Test command line parameter that shall not be localized. Same below.

File: BoostTestAdapter/Resources.resx [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+    <value>Discovering ({0}):   -&gt; [{1}]</value>
+    <comment>{0} represents the name of a test discoverer class
+{1} represents a list of source file names</comment>
+  </data>
+  <data name="DiscoveryException" xml:space="preserve">
+    <value>Exception caught while discovering tests: {0} ({1})</value>
+    <comment>{0} represents an exception message
+{1} represents an exception HResult</comment>
+  </data>
+  <data name="DiscoveryExceptionFor" xml:space="preserve">
```
Comment (LukaszMendakiewicz) [inline BoostTestAdapter/Resources.resx:167]: Unused?

**Review by LukaszMendakiewicz — DISMISSED** [review]

File: BoostTestAdapter/Resources.resx (hunk 6 above)
Comment (rpjohnst) [reply BoostTestAdapter/Resources.resx:167]: Should have been used at ListContentDiscoverer.cs:129

File: BoostTestAdapter/BoostTestExecutor.cs (hunk 2 above)
Comment (rpjohnst) [reply BoostTestAdapter/BoostTestExecutor.cs:360]: Resolved

File: BoostTestAdapter/Boost/Runner/BoostTestRunnerBase.cs (hunk 1 above)
Comment (rpjohnst) [reply BoostTestAdapter/Boost/Runner/BoostTestRunnerBase.cs]: Resolved

**Review by LukaszMendakiewicz — APPROVED** [review]

**Pull request merged**